sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestController.java

```diff
@@ -211,27 +220,38 @@ private void handleAssociatedCase(
 		final boolean equalDisease = dto.getTestedDisease() == caze.getDisease();
 
 		Runnable callback = () -> {
-			if (equalDisease && PathogenTestResultType.NEGATIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
-				showChangeAssociatedSampleResultDialog(dto, handleChanges -> {
-				});
+			if (equalDisease
+				&& PathogenTestResultType.NEGATIVE.equals(dto.getTestResult())
+				&& dto.getTestResultVerified()
+				&& !suppressSampleResultUpdatePopup) {
+				showChangeAssociatedSampleResultDialog(dto, null);
 			} else if (PathogenTestResultType.POSITIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
-				if (equalDisease) {
-					showChangeAssociatedSampleResultDialog(dto, handleChanges -> {
-						if (handleChanges) {
-							if (dto.getTestedDiseaseVariant() != null && !DataHelper.equal(dto.getTestedDiseaseVariant(), caze.getDiseaseVariant())) {
-								showCaseUpdateWithNewDiseaseVariantDialog(
-									caze,
-									dto.getTestedDiseaseVariant(),
-									dto.getTestedDiseaseVariantDetails(),
-									() -> {
-										// Retrieve the case again because it might have changed
-										showConfirmCaseDialog(FacadeProvider.getCaseFacade().getByUuid(caze.getUuid()));
-									});
-							} else {
-								showConfirmCaseDialog(caze);
-							}
+
+				// the check is supposed to be skipped when a sample result update is denied
+				Consumer<Boolean> checkForDiseaseVariantUpdate = (skip) -> {
```

Comment:
it's a bit hard to read. I would move `checkForDiseaseVariantUpdate` into a separate method without the skip param. and add the skip check in a lambda when calling `showChangeAssociatedSampleResultDialog(dto, checkForDiseaseVariantUpdate);`